### PR TITLE
Add negative log-likelihood loss

### PR DESCRIPTION
**Pull request by davidmascharka**

### CODE REVIEW

File: src/mygrad/nnet/losses/negative_log_likelihood.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import numpy as np
+
+from mygrad import Tensor, sum
```
Comment (rsokl) [inline src/mygrad/nnet/losses/negative_log_likelihood.py:3]: Nit: let's import sum as `mygrad_sum` or something like that. Just so we can't unwittingly delete this import and still have the code "work"... but super slowly

File: src/mygrad/nnet/losses/negative_log_likelihood.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    Parameters
+    ----------
+    x : array_like, shape=(N, C)
+        The C log-probabilities for each of the N pieces of data.
+
+    y_true : array_like, shape=(N,)
+        The correct class indices, in [0, C), for each datum.
+
+    weights : array_like, optional (default=None)
+        The weighting factor to use on each class, or None.
```
Comment (rsokl) [inline src/mygrad/nnet/losses/negative_log_likelihood.py:22]: This should be documented as shape-(C,), right?

File: src/mygrad/nnet/losses/negative_log_likelihood.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    >>> negative_log_likelihood(x, y_true)
+    Tensor(4.6)
+    """
+    if isinstance(y_true, Tensor):
+        y_true = y_true.data
+
+    check_loss_inputs(x, y_true)
+
+    label_locs = (range(len(y_true)), y_true)
+    factors = weights[y_true] if weights is not None else np.ones_like(y_true)
```
Comment (rsokl) [inline src/mygrad/nnet/losses/negative_log_likelihood.py:64]: Do you think we should do an input validation on the shape of `weights`: ndim-1, shape-(C,)?

**Comment by rsokl** [comment]
@davidmascharka I see these "marked as resolved" but I do not see the changes - just in case you thought you pushed them.

**Comment by davidmascharka** [comment]
heh the classic commit and don't push :flushed:

**Pull request merged**